skrub/_sentence_encoder.py

```diff
@@ -0,0 +1,300 @@
+import numbers
+import warnings
+from pathlib import Path
+
+from sklearn.base import TransformerMixin
+from sklearn.decomposition import PCA
+from sklearn.utils.validation import check_is_fitted
+
+from . import _dataframe as sbd
+from ._on_each_column import RejectColumn, SingleColumnTransformer
+from ._utils import import_optional_dependency, unique_strings
+from .datasets._utils import get_data_dir
+
+
+class ModelNotFound(ValueError):
+    pass
+
+
+class SentenceEncoder(SingleColumnTransformer, TransformerMixin):
+    """Encode string features by applying an embedding model downloaded \
+        from the HuggingFace Hub.
+
+    This is a thin wrapper around :class:`~sentence_transformers.SentenceTransformer`
+    that follows the scikit-learn API, making it usable within a scikit-learn pipeline.
+
+    .. warning::
+
+        To use this class, you need to install the optional ``transformers``
+        dependencies for skrub. For example, you can use pip:
+
+        .. code::
+
+           pip install skrub[transformers] -U
+
+        This will install the ``sentence-transformers``, ``transformers``,
+        and ``torch`` libraries on your machine. Be aware that this might lead
+        to dependency conflicts.
```

Comment:
not sure I understand the part about conflicts?

Reply:
It was to answer Gael concern's that the dependencies are constraining, for instance using pytorch < 2.4.1 with numpy > 2 leads to a segfault. Happy to change the wording or remove it entirely.

Reply:
I would say that is the case for all packages (old versions of scikit-learn are probably not compatible with numpy 2 either) so maybe not necessary to re-state here, but it's true these new ones might be more picky. I'll let you decide no strong opinion

Reply:
The only difference with dependencies like scikit-learn is that errors are raised in a manner that is easy to understand, whereas torch plainly segfault and crash the shell session :/